src/Functions/leftPadString.cpp

```diff
@@ -0,0 +1,205 @@
+#include <Columns/ColumnDecimal.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnString.h>
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+
+#include <type_traits>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+    struct LeftPadStringImpl
+    {
+        static void vector(
+            const ColumnString::Chars & data,
+            const ColumnString::Offsets & offsets,
+            const size_t length,
+            const String & padstr,
+            ColumnString::Chars & res_data,
+            ColumnString::Offsets & res_offsets)
+        {
+            size_t size = offsets.size();
+            res_data.resize((length + 1 /* zero terminator */) * size);
+            res_offsets.resize(size);
+
+            const size_t padstr_size = padstr.size();
+
+            ColumnString::Offset prev_offset = 0;
+            ColumnString::Offset res_prev_offset = 0;
+            for (size_t i = 0; i < size; ++i)
+            {
+                size_t data_length = offsets[i] - prev_offset - 1 /* zero terminator */;
+                if (data_length < length)
+                {
+                    for (size_t j = 0; j < length - data_length; ++j)
+                        res_data[res_prev_offset + j] = padstr[j % padstr_size];
+                    memcpy(&res_data[res_prev_offset + length - data_length], &data[prev_offset], data_length);
+                }
+                else
+                {
+                    for (size_t j = 0; j < length; ++j)
+                        res_data[res_prev_offset + j] = data[prev_offset + j];
```

Comment:
When it's possible it's better to use `memcpy` instead of copying in a cycle, because `memcpy` is faster.